aptos-move/block-executor/src/txn_provider/mod.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+mod blocking_txns_provider;
+pub mod default;
+
+use aptos_mvhashmap::types::TxnIndex;
+use aptos_types::transaction::BlockExecutableTransaction as Transaction;
+use std::sync::Arc;
+
+pub trait TxnProvider<T: Transaction> {
+    /// Get total number of transactions
+    fn num_txns(&self) -> usize;
+
+    /// Get a reference of the txn object by its index.
+    fn get_txn(&self, idx: TxnIndex) -> Arc<T>;
```

Comment:
Why do we need Arc's here?

Reply:
Because the` get_txn()` in `BlockingTxnsProvider` (https://github.com/aptos-labs/aptos-core/pull/14568/files#diff-1aa099ac52b6d70f8bccabd99c70a84dad9661be540fc30d7148a0047e1a945dR69) was no kind enough let me use a reference. It has a mutex object which complains when pass a reference